### ISSUE TITLE
train

### ISSUE THREAD

**Issue by uliuycy** [Y-debug-sys/Diffusion-TS#91]
Why does my training loss turn into NaN during training, and why do the generated data and the original data not overlap at all in the dimensionality reduction plot?
![Figure_1](https://github.com/user-attachments/assets/99f43873-bbba-432a-9b07-410ea852407b)
![Figure_2](https://github.com/user-attachments/assets/e05c9df7-a155-46d1-bf35-4377060dbe3c)

**Comment by Y-debug-sys** [comment]
See https://github.com/Y-debug-sys/Diffusion-TS/issues/76 and https://github.com/Y-debug-sys/Diffusion-TS/issues/6 .

**Comment by uliuycy** [comment]
> 请参阅 #76 和 #6 。

I tried using stock data several times, and the loss turns into NaN shortly after the training begins.
import os
import torch
import numpy as np

from engine.solver import Trainer
from Utils.metric_utils import visualization
from Data.build_dataloader import build_dataloader
from Utils.io_utils import load_yaml_config, instantiate_from_config
from Models.interpretable_diffusion.model_utils import unnormalize_to_zero_to_one


class Args_Example:
  def __init__(self) -> None:
    self.config_path = './Config/stocks.yaml'
    self.save_dir = './stocks'
    self.gpu = 0
    os.makedirs(self.save_dir, exist_ok=True)


args = Args_Example()
configs = load_yaml_config(args.config_path)
device = torch.device(f'cuda:{args.gpu}' if torch.cuda.is_available() else 'cpu')

dl_info = build_dataloader(configs, args)
model = instantiate_from_config(configs['model']).to(device)
trainer = Trainer(config=configs, args=args, model=model, dataloader=dl_info)

trainer.train()

dataset = dl_info['dataset']
seq_length, feature_dim = dataset.window, dataset.var_num
ori_data = np.load(os.path.join(dataset.dir, f"stock_norm_truth_{seq_length}_train.npy"))
# ori_data = np.load(os.path.join(dataset.dir, f"{dataset_name}_norm_truth_{seq_length}_train.npy"))  # Uncomment the line if dataset other than Sine is used.
fake_data = trainer.sample(num=len(dataset), size_every=2001, shape=[seq_length, feature_dim])
#fake_data = np.load('./fajiao/batch1/ddpm_fake_fj.npy')
if dataset.auto_norm:
    fake_data = unnormalize_to_zero_to_one(fake_data)
    np.save(os.path.join(args.save_dir, f'ddpm_fake_stocks.npy'), fake_data)
print(ori_data .shape,fake_data .shape)
visualization(ori_data=ori_data, generated_data=fake_data, analysis='pca', compare=ori_data.shape[0])

visualization(ori_data=ori_data, generated_data=fake_data, analysis='tsne', compare=ori_data.shape[0])

visualization(ori_data=ori_data, generated_data=fake_data, analysis='kernel', compare=ori_data.shape[0])

**Comment by uliuycy** [comment]
"After downloading your code and running `Tutorial_0`, along with the stock data provided in your link, what might be causing the issue?"

**Comment by Y-debug-sys** [comment]
Try a different random seed.

**Comment by uliuycy** [comment]
Is it normal for the loss of other random seeds to sometimes turn into nan? Is there any other way

**Comment by Y-debug-sys** [comment]
Sorry, I haven't encountered this issue myself on my own machine (both Windows and Linux). Based on previous problems, it seems to be a rare occurrence. I'll try my best to identify the cause. In the meantime, please feel free to let me know if you have any ideas.
